lib/teleterm/gateway/config.go

```diff
@@ -83,5 +88,13 @@ func (c *Config) CheckAndSetDefaults() error {
 		return trace.BadParameter("missing target URI")
 	}
 
+	if c.CLICommandProvider == nil {
+		return trace.BadParameter("missing CLICommandProvider")
+	}
+
+	if c.TCPPortAllocator == nil {
+		return trace.BadParameter("missing TCPPortAllocator")
```

Comment:
should `NetTCPPortAllocator` be the default?

Reply:
Ah, sure, I can make it so. I wasn't sure if it's a common approach to define defaults for those "collaborator" objects or if I should explicitly request them.

Reply:
Not a big deal either way. I think the question is if we see other "production" code will be using a different `TCPPortAllocator`. If so, current way is the best. If not, and the only purpose is overwriting for testing, might set a default if not provided (also in that case you can keep `tcpPortAllocator` private.)

Reply:
I'll make it the default then.